src/status_im/ui2/screens/chat/photo_selector/view.cljs

```diff
@@ -0,0 +1,107 @@
+  [rn/view
+   [rn/image {:source {:uri item}
+              :style  {:width         56
+                       :height        56
+                       :border-radius 8
+                       :margin-bottom 20}}]
+   [rn/touchable-opacity
+    {:on-press (fn [] (reset! selected (vec (filter #(-> % (not= item)) @selected))))
+     :style    (style/remove-photo-container)}
+    [quo2/icon :i/close {:color colors/white :size 12}]]])
+
+(defn bottom-gradient []
+  [:f>
+   (fn []
+     (let [safe-area (safe-area/use-safe-area)]
+       (when (> (count @selected) 0)
+         [linear-gradient/linear-gradient
+          {:colors [:black :transparent]
+           :start  {:x 0 :y 1}
+           :end    {:x 0 :y 0}
+           :style  (style/gradient-container safe-area)}
+          [rn/flat-list {:key-fn                  (fn [item] item)
+                         :render-fn               small-image
+                         :data                    @selected
+                         :horizontal              true
+                         :content-container-style {:padding-horizontal 20 :margin-top 12}
+                         :separator               [rn/view {:style {:width 12}}]}]
+          [rn/view {:style (style/buttons-container safe-area)}
+           [quo2/button {:type     :grey
+                         :style    {:flex 0.48}
+                         :on-press #(js/alert "Add text: to be implemented")}
+            (i18n/label :t/add-text)]
+           [quo2/button {:style    {:flex 0.48}
+                         :before   :send
+                         :on-press #(do
+                                      (rf/dispatch [:chat.ui/send-current-message])
+                                      (reset! selected [])
+                                      (rf/dispatch [:bottom-sheet/hide]))}
+            (str (i18n/label :t/send) " " (when (> (count @selected) 1) (count @selected)))]]])))])
+
+(defn clear-button []
+  (when (> (count @selected) 0)
+    [rn/touchable-opacity {:on-press #(reset! selected [])
+                           :style    (style/clear-container)}
+     [quo2/text {:weight :medium} (i18n/label :t/clear)]]))
+
+(defn image [item index window-width]
+  [rn/touchable-opacity
+   {:active-opacity 1
+    :on-press       (fn []
+                      (if (some #{item} @selected)
+                        (do
+                          (reset! selected (vec (filter #(-> % (not= item)) @selected)))
+                          (rf/dispatch [:chat.ui/image-unselected item]))
+                        (do
+                          (swap! selected conj item)
+                          (rf/dispatch [:chat.ui/camera-roll-pick item]))))}
+   [rn/image {:source {:uri item}
+              :style  (style/image window-width index)}]
+   (when (some #{item} @selected)
+     [rn/view {:style (style/overlay window-width)}])
+   (when (some #{item} @selected)
+     [info-count/info-count (+ (.indexOf @selected item) 1) (style/image-count)])])
```

Comment:
We should use `status-im.utils.core/first-index`. Behind the scenes it uses `keep-indexed`, which is a host-independent solution.

Reply:
I don't think it does what is needed here

Reply:
`first-index` finds the index of an element inside a collection, the only difference is you need to provide a predicate function. According to JS docs, `indexOf` returns the first index at which a given element can be found in the array, so they seem equivalent to me (except `keep-indexed` will traverse the coll in a lazy fashion).

Example:

```clojure
(first-index #(= item %) @selected)
```